src/types.ts

```diff
@@ -221,13 +217,25 @@ export type ChatCompletionOptionsFunction = {
   parameters: ObjectSchema;
 };
 
-type JSONSchema = (
-  | ObjectSchema
-  | StringSchema
-  | NumberSchema
-  | BooleanSchema
-  | ArraySchema
-) & { description?: string };
+export interface ChatCompletionMessage {
+  name?: string;
+  role: "system" | "assistant" | "user" | "function";
```

Comment:
If the `role` is `function`, is there guaranteed to be a `function_call`?

Reply:
I don't think so.  My understanding of the [limited documentation](https://platform.openai.com/docs/guides/gpt/function-calling) is that `function_call` is returned in cases where the `assistant` is asking for a function to be run.  Then, `role="function"` is used when your code wants to return the function result.  

With that in mind, I could limit `function_call` to be `assistant` specific if that makes sense.

Reply:
> With that in mind, I could limit function_call to be assistant specific if that makes sense.

Would be nice, exactly what I was thinking.

Reply:
👍 – I'll follow up soon with an update.

Reply:
This is updated in [43dd2a4](https://github.com/load1n9/openai/pull/19/commits/43dd2a4255ec5989e0a7fb57ad2346d14a953a1b) with explicit typing.